runtime/query.go

```diff
@@ -20,6 +21,29 @@ type Query interface {
 }
 
 func (r *Runtime) Query(ctx context.Context, instanceID string, query Query, priority int) error {
-	// TODO: Add caching here
-	return query.Resolve(ctx, r, instanceID, priority)
+	// take all deps and their last updated time and add them to the query key, prefix query key with instanceID
+	cacheKey := instanceID + query.Key()
+	deps := query.Deps()
+	service, err := r.catalogCache.get(ctx, r, instanceID)
+	if err != nil {
+		return err
+	}
+	for _, dep := range deps {
+		entry, found := service.FindEntry(ctx, dep)
```

Comment:
concerned about the performance penalty here as its going to hit database for fetching the dependency details, probably we can cache it but then will have to update them on reconciliation.

Reply:
perhaps we can skip the cache for small time queries that are cached on duckdb. Like cardinality and null count.

Reply:
which type of queries are cached on duckdb, is there any documentation for that ?

Reply:
do you have to enable caching on duckdb or is it on by default ?

Reply:
Even small queries ought to benefit from being cached. But we could say that if a query returns an empty cache key, then it means not to cache.

Reply:
@AdityaHegde ok you mean the table/column stats are cached at duckdb. @begelundmuller I can make that change its a good idea, if key is `nil` then don't cache it